errortypes/errortypes.go

```diff
@@ -88,6 +89,21 @@ func (err *FailedToRequestBids) Code() int {
 	return FailedToRequestBidsCode
 }
 
+// NonFatalValidation is used at the request validation step, where we want to continue processing as best we
+// can rather than returning a 4xx, and still return an error message.
+// The initial usecase is to flag deprecated bidders.
+type NonFatalValidation struct {
```

Comment:
I think this name is too vague... for example: `BadInput` errors are also a type of "non-fatal validation" errors, from the caller's point of view.

I think this should be something more specific, like `BidderTemporarilyDisabled`. Also check out the error-handling blog post from my other comment. Using that strategy, you can make this specialized and still have validation code that extends well when other non-fatal cases come up.